### DISCUSSION TITLE
Persisting state in DB

### DISCUSSION THREAD

**Discussion by arbhuiyan** [statelyai/xstate#3962] [Q&A] [answered]
Hello everyone,
i'm trying to use this package for a backend project and i need to store the state in DB. The problem i'm having is on how to initialise the state from a specific state (persisted state). There is an example on docs on how to store the state in browser and it seems to require entire Store object not only the state key.
Can you please advise if there a good approach for this case?

Thanks in advance

**Comment by davidkpiano** [comment, marked as answer]
At minimum, you only need to store the `state.value` and `state.context` if applicable.

Then, you can recreate the state (XState v4 code):

```ts
import { State } from 'xstate';

const someMachine = createMachine({...});

// ... get value and context from DB

const restoredState = State.from(value, context);

// Resolves the state so that it is compatible with someMachine
const resolvedState = someMachine.resolveState(restoredState);

interpret(someMachine).start(resolvedState);
```
